### DISCUSSION TITLE
03_pytorch_computer_vision_video

### DISCUSSION THREAD

**Discussion by hardery** [mrdbourke/pytorch-deep-learning#123] [Q&A]
hello, mrdbourke
this is the code around at around 15:12:00
`from torch.utils.data import DataLoader

BATCH_SIZE = 32

train_dataloader = DataLoader(dataset=train_data, batch_size=BATCH_SIZE, shuffle=True)

test_dataloader = DataLoader(dataset=test_data, batch_size=BATCH_SIZE, shuffle=True)

train_dataloader, test_dataloader`

NOTICE:  I have set the shuffle=True of  test_dataloader which you set the value False

then I get the confmat_tensor matrix like this at around 19:24:00:
![无标题](https://user-images.githubusercontent.com/51631808/189550093-045ca406-ca79-4073-bcab-7d7abe214d06.png)

here is my doubts:
1. I set the shuffle=True of the test_dataloader, and the accuracy seems fine to me.
Epoch: 0
Train Loss: 0.5883, Train Acc:78.78%
Test Loss: 0.3993, Test Acc: 86.15%

Epoch: 1
Train Loss: 0.3657, Train Acc:86.88%
Test Loss: 0.3494, Test Acc: 87.49%

Epoch: 2
Train Loss: 0.3295, Train Acc:88.10%
Test Loss: 0.3224, Test Acc: 88.19%
Train time on None: 37.60443163400001 seconds
37.60443163400001

2.  I got no error in the training or eval process. but the confmat_tensor matrix makes me feel confusion. But when I change shuffle=False, the confmat_tensor matrix seems OK. Why is that???

**Comment by hardery** [comment]
>you can test it by setting shuffle=True of the test_dataloader, then you will get a confmat_tensor matrix.
>I have looked at the document of pytorch, it says if shuffle=True, then it pick up values randomly in the "for loop".
